### PR TITLE
feat(import): add gitlab annotation

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/KFLUXBUGS-1440

## Description

Adds Gitlab annotations for GitLab component creation

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2024-09-16 at 5 59 01 PM

Screenshot 2024-09-30 at 4 09 41 PM
Screenshot 2024-09-30 at 4 09 16 PM

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by sahil143** [comment]
/hold

until we have a clear indication of what is needed on UI.

File: src/components/ImportForm/ComponentSection/ComponentSection.tsx [hunk 1]
```diff
@@ -28,6 +76,16 @@ export const ComponentSection = () => {
...
+      <ExpandableSection
+        toggleTextExpanded="Hide advanced annotation options"
+        toggleTextCollapsed="Show advanced annotation options"
+        isExpanded={isAnnotationExpanded}
+        data-test="expand-annotation-section"
+        onToggle={() => setAnnotationExpanded((x) => !x)}
+      >
+        <GitProviderDropdown name="gitProviderAnnotation" />
+        <InputField name="gitURLAnnotation" label="Git url annotation" data-test="url-annotation" />
+      </ExpandableSection>
```
Comment (sahil143) [inline src/components/ImportForm/ComponentSection/ComponentSection.tsx:88]: We don't need a separate expandable section, you can move these two fields under the Advanced git options

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/components/ImportForm/ComponentSection/ComponentSection.tsx (hunk 1 above)
Comment (abhinandan13jan) [reply src/components/ImportForm/ComponentSection/ComponentSection.tsx]: updated

**Comment by Katka92** [comment]
This PR needs a change in tests. Will you fix it @abhinandan13jan ? It's about fixing dropdown locator for `Pipeline` item [here](https://github.com/openshift/hac-dev/blob/main/integration-tests/support/pageObjects/createApplication-po.ts#L50) as now one more is added.

**Comment by sahil143** [comment]
> This PR needs a change in tests. Will you fix it @abhinandan13jan ? It's about fixing dropdown locator for `Pipeline` item [here](https://github.com/openshift/hac-dev/blob/main/integration-tests/support/pageObjects/createApplication-po.ts#L50) as now one more is added.

@abhinandan13jan Can you please fix the failing e2e test?

**Comment by abhinandan13jan** [comment]
@Katka92 Thanks for the e2e fix.
@sahil143 You can now proceed with the review

**Comment by Misjohns** [comment]
@abhinandan13jan Can we prefill the Git provider annotation from the Git repo URL?

**Comment by abhinandan13jan** [comment]
> we prefill the Git provider annotation from the Git repo URL?

@Misjohns Yes we are auto populating based on the URL

**Comment by Misjohns** [comment]
LGTM

**Review by sahil143 — APPROVED** [review]
/lgtm
/approve

**Pull request merged**